lld/ELF/Options.td

```diff
@@ -141,6 +141,20 @@ def call_graph_profile_sort: JJ<"call-graph-profile-sort=">,
 def : FF<"no-call-graph-profile-sort">, Alias<call_graph_profile_sort>, AliasArgs<["none"]>,
   Flags<[HelpHidden]>;
 
+defm irpgo_profile_sort: Eq<"irpgo-profile-sort",
```

Comment:
On the embedded side, we've discussed consuming profiles for an alternative goal: automatically placing hotter sections in better regions of memory.

Would it be possible to separate out the flag that provides the profile, say, `--irpgo_profile`, from a flag that enables sorting for startup time, say, `--startup-sort`? There's already a small family of sort concerns present in this patch, and that organization would support adding the embedded one in-kind once we iron out the details.

Reply:
LGTM. 
How about this separation in Mach-O, as it is currently used in both ELF and Mach-O? Or implement it just in ELF in this commit?

Reply:
Ideally implement the change for Mach-O in a prerequisite patch, then implement it in this PR as a follow-up

guideline: improve the base, then port the base

Reply:
@MaskRay @ellishg @mysterymath I have created a new pull request about the options change. Could you please review it?

https://github.com/llvm/llvm-project/pull/118594
If it works I'll rebase this one.